hail/src/test/scala/is/hail/expr/ir/RequirednessSuite.scala

```diff
@@ -519,3 +519,22 @@ class RequirednessSuite extends HailSuite {
     assert(actual == expected)
   }
 }
+
+object RequirednessSuite {
+  def deepInnerRequired(t: PType, required: Boolean): PType =
+    t match {
+      case t: PArray => PCanonicalArray(deepInnerRequired(t.elementType, true), required)
```

Comment:
can we change this while we're here? It's wrong -- should match on the same types (PCArray, PCSet, PCDict, PCStruct.

Reply:
of course